src/main/scala/viper/gobra/frontend/Parser.scala

```diff
@@ -873,7 +873,21 @@ object Parser {
 
     lazy val predeclaredType: Parser[PPredeclaredType] =
       "bool" ^^^ PBoolType() |
-        "int" ^^^ PIntType()
+        // signed integer types
+        "rune" ^^^ PInt32Type() |
+        "int8" ^^^ PInt8Type() |
+        "int16" ^^^ PInt16Type() |
+        "int32" ^^^ PInt32Type() |
+        "int64" ^^^ PInt64Type() |
+        "int" ^^^ PIntType() | // 'int' must come after all 'intX' parsers, otherwise parsing will stop at 'int'
+        // unsigned integer types
+        "byte" ^^^ PUInt8Type() |
+        "uint8" ^^^ PUInt8Type() |
```

Comment:
Since both are parsed as the same type, the type checker is not able to distinguish between byte and uint8. This is only ok, if these types are indistinguishable in their behaviour. Is this the case here? Similarly, for unes and int32, etc.

Reply:
Yes, the Go spec defines byte and rune as aliases to uint8 and int32 respectively (https://golang.org/ref/spec#Numeric_types). As an additional check, I verified that
```go
func main() {
	var a byte = uint8(1)
	var b uint8 = byte(2)
	fmt.Printf("%d %d\n", a, b)

	var c int32 = rune(1)
	var d rune = int32(1)
	fmt.Printf("%d %d\n", c, d)
}
```
compiles without type errors and warnings.